src/os/bluestore/BlueStore.cc

```diff
@@ -1596,7 +1596,30 @@ bool BlueStore::ExtentMap::update(Onode *o, KeyValueDB::Transaction t,
     }
     // will persist in the onode key, see below
   } else {
+
+    struct dirty_shard_t{
+    string* key = nullptr;
+    bufferlist bl;
```

Comment:
If we put the bufferlist dirty_bl in Shard, we can avoid this structure entirely, right?